src/realm/sync/changeset_parser.cpp

```diff
@@ -486,8 +486,7 @@ void State::parse_one()
         case Instruction::Type::Clear: {
             Instruction::Clear instr;
             read_path_instr(instr);
-            uint32_t prior_size = read_int<uint32_t>();
-            static_cast<void>(prior_size); // Ignored
+            instr.collection_type = read_collection_type();
```

Comment:
Do we want to validate that for the purposes of a Clear instruction a `Single` collection type is not valid?

Reply:
the InstructionApplier will do that (in an upcoming pr), but it can be done here too

Reply:
I think if it's not too much code let's do it here.

Reply:
it's definitely not

Reply:
@jbreams if we make this change, our tests will fail because the server does not send the collection type yet
@tkaye407, @niharika01 any eta of when the server work is going to be merged?

Reply:
Ah, got it, then can you add a `//TODO` here so we don't forget? And we can fix that in a follow-on commit.